agent/src/capability/baseconfigstore.cpp

```diff
@@ -204,11 +204,19 @@ std::vector<std::string> BaseConfigStore::fileList(const std::string &dirPath)
     return files;
 }
 bool BaseConfigStore::isJsonFile(const std::string &filename) {
-    size_t lastdot = filename.find_last_of(".");
-    std::string prefix = filename.substr(lastdot);
-    if (prefix.compare(".json") != 0) {
-        log_debug() << "File does not have json as prefix (" << prefix <<")";
+
+    if (filename.size() < 5) {
         return false;
     }
+
+    std::string prefix = ".json";
+    size_t lastFivePos = filename.size()-5;
+
+    std::string lastFiveLetters = filename.substr(lastFivePos);
+
+    if (prefix.compare(lastFiveLetters) != 0) {
+        return false;
+    }
+
```

Comment:
This is a very roundabout way to check if the end is ".json". How about this:
`return filename.substr(filename.size()-5) == ".json"`
I guess the check for string length is reasonable though.